### ISSUE TITLE
More FPS, by 'show()'ing less...

### ISSUE THREAD

**Issue by mungewell** [samveen/pico-oled-1.3-driver#1]
I have a project which needs a fast display, for showing counter at high rate. I figured somewhat of a solution that might be of interest.

I overload the `show()` function to specify a block of lines which are to be update (rest of screen left 'as-is'). It seems that the I2C/SPI is quite slow, and a large factor in duration of drawing the screen.
```
class FastShow(OLED_1inch3_SPI):
    def show(self, start=0, end=-1):
        # Allow user to limit show to range of rows, speeding up display
        line = start
        if end < 0:
            end = self.height
        
        if end < line:
            line = end
            end = start

        self.write_cmd(0xb0)
        for page in range(line,end):
            self.column = 63 - page
            self.write_cmd(0x00 + (self.column & 0x0f))
            self.write_cmd(0x10 + (self.column >> 4))
            for num in range(0,16):
                self.write_data(self.buffer[page*16+num])
```
Demo:
[blit_test.py.txt](https://github.com/samveen/pico-oled-1.3-driver/files/14200204/blit_test.py.txt)

Might be a good improvement to add to your driver...

**Comment by samveen** [comment]
@mungewell Thank you for this suggestion. It would indeed help a lot with display refreshing speeds.

Might I interest you to create a pull request, as I might not be able to do as good a job as you with this?

**Comment by mungewell** [comment]
Happy to do a pull request...

After quiet a bit of experimentation I figured you can also limit the `show()` in the X direction too, so this improved demo only updates the characters of the count which have changed... managing to get ~88fps (over twice as fast as before).
[blit_test.py.txt](https://github.com/samveen/pico-oled-1.3-driver/files/14205012/blit_test.py.txt)

It's a little more convoluted, so sticking with just spec'ing the lines to update is probably easier for the 'pull'.

**Comment by samveen** [comment]
Fixed by #2 from @mungewell
